Java/policy-service/src/main/java/com/insuranceapplication/policyservice/methods/Utils.java

```diff
@@ -1,23 +1,146 @@
+        ResponseEntity response = template.postForEntity(eurekaClient.getApplication(Variables.dbName)
+                .getInstances().get(0).getHomePageUrl() + "/getpolicyline", policy, PolicyLine.class);
+        PolicyLine policyLine = (PolicyLine) response.getBody();
+        List<InsuredObject> insObjects = Utils.mapToList((List<LinkedHashMap>) policyService.getInsuredObjects(policyLine), InsuredObject.class);
+        return insObjects.stream().filter(x -> x.getType().equals(objectType)).collect(Collectors.toList()).get(0);
+    }
+
+    /**
+     * helper method to get correct configuration variables.
+     *
+     * @param combinationName
+     * @param comboId
+     * @return returns configuration row of params.
+     */
+    public PremiumCalcConfigValue getCalcConfigValue(String combinationName, String comboId, String version) {
+        calcVariables = Utils.mapToList((List<LinkedHashMap>) policyService.getAllPremiumValuesConfig().getBody(), PremiumCalcConfigValue.class);
+        return calcVariables.stream()
+                .filter(x -> x.getCombinationName().equals(combinationName) && x.getComboId().equals(comboId) && x.getVersion().equals(version))
+                .collect(Collectors.toList()).get(0);
+    }
+
+    public PremiumCalcConfigValue getCalcConfigValue(String combinationName, String comboId, String version, int age) {
+        calcVariables = Utils.mapToList((List<LinkedHashMap>) policyService.getAllPremiumValuesConfig().getBody(), PremiumCalcConfigValue.class);
+        OptionalInt maxValue = calcVariables.stream()
+                .filter(x -> x.getCombinationName().equals(combinationName) && x.getComboId().equals(comboId) && x.getVersion().equals(version))
+                .collect(Collectors.toList())
+                .stream().mapToInt(x -> Integer.parseInt(x.getValue2())).max();
+        OptionalInt minVal = calcVariables.stream()
+                .filter(x -> x.getCombinationName().equals(combinationName) && x.getComboId().equals(comboId) && x.getVersion().equals(version))
+                .collect(Collectors.toList())
+                .stream().mapToInt(x -> Integer.parseInt(x.getValue2())).min();
+        PremiumCalcConfigValue recordWithLesserValue = calcVariables.stream()
+                .filter(x -> x.getVersion().equals(version) && x.getCombinationName().equals(combinationName) && x.getComboId().equals("LBE") && Integer.parseInt(x.getValue2()) == minVal.getAsInt()).collect(Collectors.toList()).get(0);
+        PremiumCalcConfigValue recordWithBiggerValue = calcVariables.stream()
+                .filter(x -> x.getVersion().equals(version) && x.getCombinationName().equals(combinationName) && x.getComboId().equals("LBE") && Integer.parseInt(x.getValue2()) == maxValue.getAsInt()).collect(Collectors.toList()).get(0);
+        return (age >= maxValue.getAsInt()) ? recordWithBiggerValue : recordWithLesserValue;
+    }
+
+    public double findInsuranceSumForCertainRisk(String riskId, String version) {
```

Comment:
Takie metody nie powinny znajdować się w klasie przechowującej metody ogólne. To powinno zawierać się w miejscu w którym to będzie wykorzystane. Tu ewidentnie widać że będzie to wykorzystywane tylko przez klasę PremiumCalculation.